src/main/java/uk/co/ramp/policy/SingleCaseIsolationPolicy.java

```diff
@@ -0,0 +1,93 @@
+package uk.co.ramp.policy;
+
+import org.immutables.value.Value;
+import uk.co.ramp.distribution.Distribution;
+import uk.co.ramp.distribution.DistributionSampler;
+import uk.co.ramp.distribution.ImmutableDistribution;
+import uk.co.ramp.people.AlertStatus;
+import uk.co.ramp.people.VirusStatus;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import static java.util.Comparator.comparingInt;
+import static uk.co.ramp.distribution.ProgressionDistribution.FLAT;
+
+class SingleCaseIsolationPolicy {
+    private final IsolationProperties isolationProperties;
+    private final DistributionSampler distributionSampler;
+    private final Distribution infinityDistribution = ImmutableDistribution.builder()
+            .type(FLAT)
+            .mean(Double.MAX_VALUE)
+            .max(Double.MAX_VALUE)
+            .build();
+
+    @Value.Immutable
+    interface IsolationMapValue {
+        int startTime();
+        int maxIsolationTime();
+        IsolationProperty isolationProperty();
+    }
+
+    private final Map<Integer, IsolationMapValue> currentlyInIsolationMap = new HashMap<>();
+
+    SingleCaseIsolationPolicy(IsolationProperties isolationProperties, DistributionSampler distributionSampler) {
+        this.isolationProperties = isolationProperties;
+        this.distributionSampler = distributionSampler;
+    }
+
+    private IsolationProperty findRelevantIsolationProperty(double actualInfectedProportion, VirusStatus virusStatus, AlertStatus alertStatus) {
+        Stream<IsolationProperty> globalIsolationPolicy = isolationProperties.globalIsolationPolicies().values().stream()
+                .filter(policy -> policy.proportionInfected().min() < actualInfectedProportion * 100)
+                .filter(policy -> policy.proportionInfected().max() > actualInfectedProportion * 100)
+                .map(ProportionInfectedIsolationProperty::isolationProperty);
+
+        Stream<IsolationProperty> virusIsolationPolicy = isolationProperties.individualIsolationPolicies().virusStatusPolicies().values().stream()
+                .filter(policy -> policy.virusStatus() == virusStatus)
+                .map(VirusStatusIsolationProperty::isolationProperty);
+
+        Stream<IsolationProperty> alertIsolationPolicy = isolationProperties.individualIsolationPolicies().alertStatusPolicies().values().stream()
+                .filter(policy -> policy.alertStatus() == alertStatus)
+                .map(AlertStatusIsolationProperty::isolationProperty);
+
+        Stream<IsolationProperty> defaultIsolationPolicy = Stream.of(isolationProperties.defaultPolicy());
+
+        return Stream.of(globalIsolationPolicy, virusIsolationPolicy, alertIsolationPolicy, defaultIsolationPolicy)
```

Comment:
As this is relying on user inputs, it might be worth adding a check to ensure that policies don't have the same priority so we don't get a clash here.

Reply:
@sibylleM As the FMD model was in Java 7, have you used streams much? If so we can provide comments as descriptions.

Reply:
Sure, I'll add in a check